02-Context-or-Narrative/State-of-PureScript-Documentation-2019.md

```diff
@@ -1,28 +1,37 @@
 # State of PureScript Documentation: 2019
 
+<!-- https://magnetikonline.github.io/markdown-toc-generate/ -->
+
+- [Why Read This Document?](#why-read-this-document)
+- [What is "Good" Documentation Anyway?](#what-is-good-documentation-anyway)
+    - [The Types of Documentation](#the-types-of-documentation)
+    - [The Documentation's Intended Audience](#the-documentations-intended-audience)
+    - [Maintaining Documentation's Accuracy](#maintaining-documentations-accuracy)
+        - [The "Size" of a Change](#the-size-of-a-change)
+        - [The "Frequency" of a Change](#the-frequency-of-a-change)
+        - [How to Make Maintenance Easier](#how-to-make-maintenance-easier)
+    - [Criteria for "Good" Documentation](#criteria-for-good-documentation)
+    - [Evaluating PureScript's Documentation](#evaluating-purescripts-documentation)
+- [Why is PureScript's Documentation Lacking and How Do We Improve It?](#why-is-purescripts-documentation-lacking-and-how-do-we-improve-it)
+- [New Learners: What is the Best Way to Learn PureScript?](#new-learners-what-is-the-best-way-to-learn-purescript)
+- [PureScript Documentation Writers: What is the Best Way to Write Documentation in this Context?](#purescript-documentation-writers-what-is-the-best-way-to-write-documentation-in-this-context)
+- [Core Contributors: ???](#core-contributors-)
+
+
 ## Why Read This Document?
 
-This document is not meant to convince you to use PureScript. We assume that readers already believe in using PureScript instead of alternatives.
+This document is not meant to convince you to use PureScript. We assume that readers already want to learn to use PureScript.
 
-It is well-known among the PureScript community that its documentation is lacking in some critical areas. As a result, @chexxor started the [PureScript documentation efforts in 2019](https://discourse.purescript.org/t/purescript-documentation-efforts-in-2019/524) discourse thread to answer one question:
+It is well-known amongst the PureScript community that its documentation is lacking in some critical areas. As a result, @chexxor started the [PureScript documentation efforts in 2019](https://discourse.purescript.org/t/purescript-documentation-efforts-in-2019/524) Discourse thread to answer one question:
 > How can we improve the PureScript documentation in 2019?
 
-(Purpose) The following document tries to do two things:
-1. Provide an answer to @chexxor's above question
-2. Answer other related questions that various audiences will have:
-    - new learners (regardless of language background)
+(Purpose) The following document tries to do the following things:
+1. Determine why PureScript's documentation is seen as lacking.
+2. Understand the expectations & needs of the different documentation consumers.
+    - New learners (regardless of language background)
```

Comment:
"Determine" rather than "Provide" seems to be more of an investigation rather than a "Hey, there's fire in that room over there, so here's how to navigate through it." I also think the 'seen as lacking' is a bit misleading. People already believe it is lacking. That rendering makes it sound like we're going to argue why it's not actually lacking.

As for the second point's rendering, perhaps it could use "Summarize" rather than "understand." The latter makes it sound like we're going to ask questions rather than say, "Here's the gist of things." I like the 'expectations and needs' part. 
I think the 'documentation consumers' part is a bit narrow because of the 'consumer' part. It seems like it's only addressed to new learners (who consume docs) rather than a wider audience (some of who may write/produce docs and others who affect the documentation situation).

Reply:
Good ideas, I'll change it to that point of view.

The reason I wanted to add some qualification around "lacking" is because I believe that's fundamentally an opinionated thing. A thing is always seen as lacking by someone, even if 99% of people think its perfect. Even within the group of people who believe it is perfect, they are likely to change their mind when they are having problems understanding how to write an app which has a unique problem that doesn't have documentation or guidance because it is so unique.

So, I think "lacking" is unavoidable, but it *is* feasible to bring the ratio of people who feel its lacking to a more healthy number. For example, 

    - New learners - 10% of audience is satisfied
    - PureScript documentation writers - 50% of audience is satisfied
    - Core contributors - 60% of audience is satisfied

If these figures are current ratios of the satisfied consumers of the documentation, then perhaps a good goal is 75% or 85% for each of those audiences.

Reply:
I added a commit which updates the Purpose section. I don't know if it correctly addresses your concerns, but I think it has a more clear break-down. I think it needs another review - I'll certainly review it again later myself.

Reply:
> The reason I wanted to add some qualification around "lacking" is because I believe that's fundamentally an opinionated thing. A thing is always seen as lacking by someone, even if 99% of people think its perfect.

Good points! I assume we'd change `Determine` to `Explain` then? Or how would you render it now? 
Right now (with the 'explain' word used instead of 'determine'), I read this as "we'll be explaining what each relevant party (new learners, users, core contributors) thinks of the situation" rather than "we'll be explaining what are the issues behind why it's bad, regardless of who thinks that." (AFAIK, every relevant party thinks its bad)

> If these figures are current ratios of the satisfied consumers of the documentation, then perhaps a good goal is 75% or 85% for each of those audiences.

That sounds like a good goal. How would it be quantified? What does a 100% for new learners mean or look like? How is it different from a 100% for a core contributor?

Reply:
I imagine a 100% score would mean that a person from that audience will find an answer to their PureScript-related question 100% of the time. Another way of saying this, perhaps, is that documentation contains answers for 100% of the PureScript-related questions a PureScript audience has.

So, to think a bit further, a 20% score means that the documentations answers 20% of the questions a person has, regardless of how common those questions are. I guess if 1 really common problem is documented well, it would raise the score equally as much as documenting 10 niche questions well would raise the score.

Reply:
I wonder if it's worth finding a place in this project to document this idea of a "documentation score". Having metrics can be a double-edged sword: it can increase focus, but if a metric is inappropriately defined or measures the wrong thing then the project moves in a slightly wrong direction and doesn't serve its audience as effectively as it could. (I hear that often the bad side of metrics is people gaming them for personal gain, but I suspect that wouldn't be a problem in this situation, so I say 'moves it in a slightly wrong direction'.)

Reply:
Good points. It sounds like you're describing something like this:
- Findable: person can find answers to their questions (in ideally 5 minutes or less of Googling)
- Coverage: most common questions are well answered (niche ones might not be worth it as much, or just direct people to ask on the Slack chatroom)

--wait, isn't this just redescribing what we've already learned from the 'how to write good docs' section?

Reply:
> Having metrics can be a double-edged sword

Very true. They're useful when they're useful and painful in practically all other situations.

Reply:
I don't recall seeing this exact thing in the stuff I read, but it's likely that I overlooked it. :)

Reply:
Also yeah, it's probably best to keep the metric idea unmeasured for now. :)

Reply:
> I don't recall seeing this exact thing in the stuff I read, but it's likely that I overlooked it. 

I don't think it is, but one point (docs should be easily navigatable) implies things like the above